hub/core/meta/encode/chunk_id.py

```diff
@@ -0,0 +1,235 @@
+from hub.util.exceptions import ChunkIdEncoderError
+import hub
+from hub.core.storage.cachable import Cachable
+from io import BytesIO
+from typing import Optional, Tuple
+import numpy as np
+from uuid import uuid4
+
+
+CHUNK_ID_INDEX = 0
+LAST_INDEX_INDEX = 1
+
+
+class ChunkIdEncoder(Cachable):
+    def __init__(self):
+        """Encodes chunk IDs such that they can be mapped to and from sample indices.
+
+        Note:
+            This map has a time complexity of `O(log(N))`, where `N` is the number of unique chunk IDs in this instance.
```

Comment:
This should perhaps be somewhere else in the docs and not in the class docstring